### PR TITLE
Change Helm to make deploying restxds optional

### PR DESCRIPTION
**Pull request by npolshakova**
# Description

Gloo gives the option to turn on restxds, and this is needed if you want to dynamically turn restxds on/off without redeploying. This changes the Helm to make restxds deployment optional. When `enableRestEds` is true, the rest_xds_cluster will be included.

# Context

https://github.com/solo-io/gloo/issues/4164

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/4164

### CODE REVIEW

File: install/helm/gloo/templates/9-gateway-proxy-configmap.yaml [hunk 1]
```diff
@@ -187,6 +187,7 @@ data:
                     address: 127.0.0.1
                     port_value: 8234
 {{- end }}
+{{- if $.Values.settings.enableRestEds }}
```
Comment (saiskee) [inline install/helm/gloo/templates/9-gateway-proxy-configmap.yaml:190]: Out of curiosity, what is the difference between `$.Values` and `.Values`?

File: install/test/helm_test.go [hunk 2]
```diff
@@ -1899,7 +1939,7 @@ spec:
   gloo:
     xdsBindAddr: "0.0.0.0:9977"
     restXdsBindAddr: "0.0.0.0:9976"
-    enableRestEds: false
+    enableRestEds: true
```
Comment (saiskee) [inline install/test/helm_test.go:1942]: is this change needed?

Comment (saiskee) [reply install/test/helm_test.go:1942]: As in, will it affect the test? I suppose it doesn't hurt much having it, but just curious.

Comment (npolshakova) [reply install/test/helm_test.go:1942]: So it doesn't affect the test, but Ashley's PR would have conflicts (https://github.com/solo-io/gloo/pull/4155) and to get the tests passing I would need to remove the `rest_xds_cluster` from all the yaml.

File: install/helm/gloo/templates/9-gateway-proxy-configmap.yaml (hunk 1 above)
Comment (npolshakova) [reply install/helm/gloo/templates/9-gateway-proxy-configmap.yaml:190]: In this case, I just tried keeping it the same as `Values` was already being used in the yaml. But in general I thought the `.` means data getting passed into the gotemplate, versus `$` refers to a variable defined and saved in the template.

Comment (kdorosh) [reply install/helm/gloo/templates/9-gateway-proxy-configmap.yaml:190]: `.` is the variable for the current scope. `$` is a variable for the top level scope. In most cases, `$.Values` is the same as `.Values`, but some helm statements (e.g. `range`) change the current scope, which would make these values different

File: changelog/v1.7.0-beta9/optional-rest-xds.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+changelog:
+  - type: NEW_FEATURE
```
Comment (kdorosh) [inline changelog/v1.7.0-beta9/optional-rest-xds.yaml:2]: let's call this a fix

Comment (kdorosh) [reply changelog/v1.7.0-beta9/optional-rest-xds.yaml:2]: there's also a changelog of type `HELM` that's even a slightly better fit here

File: install/test/helm_test.go (hunk 2 above)
Comment (kdorosh) [reply install/test/helm_test.go:1942]: let's have ashley's PR merge first so this diff is simpler and more direct

**Review by kdorosh — APPROVED** [review]
